packages/byline-parser/index.js

```diff
@@ -34,7 +34,8 @@ export default function (blobSource, { openingPattern, closingPattern, matchPatt
         throw new Error('No capture group is defined');
       }
 
-      tmp.push(captureGroup.slice(1));
+      if (!noGroup) tmp.push(captureGroup.slice(1));
+      if (noGroup) results.push(captureGroup.slice(1));
```

Comment:
Do the opening and closing pattern need to be optional for Go? I'm wondering if we can just get rid of this commit entirely. Of course, even if Go doesn't need this functionality, it might be useful later.

In any case, these two lines could be combined into:

```js
(noGroup ? results : tmp).push(captureGroup.slice(1));
```